### PR TITLE
Initial implementation of repository subcommand

### PR DESCRIPTION
**Pull request by tboerger**

### CODE REVIEW

**Comment by vuntz** [comment]
FWIW, this will need to be updated once we add the support for multiple architectures.

**Comment by tboerger** [comment]
> FWIW, this will need to be updated once we add the support for multiple architectures.

Yeah, sure... but the basics are already correct.

File: lib/crowbar/client/command/repository.rb [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            parent.arg :repo
+            parent.command "activate" do |c|
+              c.action do |global, opts, args|
+                platform = args.shift
+                repo = args.shift
+
+                Request.instance.repository_activate(platform, repo) do |request|
+                  case request.code
+                  when 200
+                    say "Activated successfully #{repo} on #{platform}"
```
Comment (MaximilianMeister) [inline lib/crowbar/client/command/repository.rb]: As @vuntz already said, "Successfully activated" would be correct english :-)

**Comment by MaximilianMeister** [comment]
other than that :+1: 

we also have to somehow make accessible in the databag if the repo is active which could further influence the code here

**Comment by aplanas** [comment]
Can I suggest to create unit test for new code in crowbar? I really dislike grow the codebase in this way.

**Comment by tboerger** [comment]
> we also have to somehow make accessible in the databag if the repo is active which could further influence the code here

Not in the data bag but in the json response. That should be an extension later on.

**Comment by tboerger** [comment]
> Can I suggest to create unit test for new code in crowbar? I really dislike grow the codebase in this way.

This is not increasing the crowbar core codebase, it's a standalone thing. But anyway I still have it on my plan to write cucumber and rspec tests for a full test coverage.

**Comment by rsalevsky** [comment]
+1

**Pull request merged**